include/nbl/builtin/glsl/ext/FFT/default_compute_fft.comp

```diff
@@ -78,41 +56,54 @@ struct nbl_glsl_ext_FFT_output_t
 #define _NBL_GLSL_EXT_FFT_OUTPUT_DESCRIPTOR_DEFINED_
 layout(set=_NBL_GLSL_EXT_FFT_OUTPUT_SET_DEFINED_, binding=_NBL_GLSL_EXT_FFT_OUTPUT_BINDING_DEFINED_) restrict buffer OutputBuffer
 {
-	nbl_glsl_ext_FFT_output_t outData[];
+	nbl_glsl_complex outData[];
 };
 #endif
 
 // Get/Set Data Function
 
+layout(push_constant) uniform PushConstants
+{
+	layout (offset = 0) nbl_glsl_ext_FFT_Parameters_t params;
+} pc;
+
+ nbl_glsl_ext_FFT_Parameters_t nbl_glsl_ext_FFT_getParameters() {
+	 nbl_glsl_ext_FFT_Parameters_t ret;
+	 ret = pc.params;
+	 return ret;
+ }
+
 nbl_glsl_complex nbl_glsl_ext_FFT_getData(in uvec3 coordinate, in uint channel)
 {
-	vec2 retValue = vec2(0, 0);
+	nbl_glsl_complex retValue = nbl_glsl_complex(0, 0);
 #if USE_SSBO_FOR_INPUT > 0
-	uvec3 dimension = pc.dimension;
+	uvec3 dimension = nbl_glsl_ext_FFT_getDimensions();
 	uint index = channel * (dimension.x * dimension.y * dimension.z) + coordinate.z * (dimension.x * dimension.y) + coordinate.y * (dimension.x) + coordinate.x;
-	retValue = inData[index].complex_value;
+	retValue = inData[index];
 #else
 	vec4 texelValue= texelFetch(inputImage, ivec2(coordinate.xy), 0);
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/63#discussion_r575705439

Reply:
I actually tried textureLod, it worked with the "clamp_to_border" sampler as expected with dark edges.
I used the "prev_texture_coordinates/textureSize" as the coordinates to load from the image
but "clamp_to_edge" sampler just gave me a fully black coloured result. 
I didn't investigate further. but I'll test that again after everything else is set up.

Reply:
btw when you convert integer coordinates you need to do this
`(vec3(integerCoords)+vec3(0.5))/vec3(textureSize)`

This is because pixels are sampled at centers

P.S. Make sure you dont divide by zero for the z coordinate

Reply:
ohhhh nice!

Reply:
ok that's done now